programs/steward/src/state/steward_state.rs

```diff
@@ -550,6 +550,12 @@ impl StewardState {
         self.validators_for_immediate_removal
             .set(num_pool_validators, false)?;
 
+        if marked_for_regular_removal {
+            self.validators_to_remove.set(index, false)?;
+        } else {
+            self.validators_for_immediate_removal.set(index, false)?;
+        }
```

Comment:
Need to remove this and double check that no other legacy changes are made in this PR

Reply:
10-4 will check